### PR TITLE
add -a/--all option to annotatevariants intervals

### PR DESCRIPTION
**Pull request by cseed**
Broke VSM.annotateIntervals into two functions (presence vs value).

I think this resolves https://github.com/broadinstitute/hail/issues/451, although there could be some dispute.

### CODE REVIEW

File: src/main/scala/org/broadinstitute/hail/driver/AnnotateVariantsBed.scala [hunk 1]
```diff
@@ -28,9 +28,16 @@ object AnnotateVariantsBed extends Command {
...
-      vds = state.vds.annotateIntervals(t, o,
-        Parser.parseAnnotationRoot(options.root, Annotation.VARIANT_HEAD)))
+    val vds = state.vds
+    val path = Parser.parseAnnotationRoot(options.root, Annotation.VARIANT_HEAD)
+    val newVDS = BedAnnotator(options.input, state.hadoopConf) match {
+      case (is, None) =>
+        vds.annotateIntervals(is, path)
+
+      case (is, Some((t, m))) =>
+        vds.annotateIntervals(is, t, m, all = false, path)
```
Comment (tpoterba) [inline src/main/scala/org/broadinstitute/hail/driver/AnnotateVariantsBed.scala]: Why not expose --all in the BED annotator too?

**Comment by tpoterba** [comment]
One comment, back to you.

File: src/main/scala/org/broadinstitute/hail/driver/AnnotateVariantsBed.scala (hunk 1 above)
Comment (cseed) [inline src/main/scala/org/broadinstitute/hail/driver/AnnotateVariantsBed.scala]: Good suggestion.  Will add.

**Comment by cseed** [comment]
Done, back to you.

**Pull request merged**